### ISSUE TITLE
AWS Tapes ..

### ISSUE THREAD

**Issue by gavineadie** [simh/simh#910]
This is not an "issue" (nothing needs fixing), but a question .. a simple start but, maybe more complex follow up, if deemed worth it.

Magtapes images can be formatted as "AWS" tapes:

> SIMH can read and write AWS format tape images.  AWS format uses a 3 16b (little endian) word record header which are a next record size, previous record size and a flag word.  The flag word uses value 0x40 to flag tape marks, and 0xA0 for data records.

AWS tape images have been around for a very long time, used by IBM, at least back to 2001.  Formal documentation of their format, as far as I can determine, does not exist, though informal descriptions do.  I am wondering if the `simh` AWS formatting was derived from something more formal than I have found?

*This question may get more complex because I have an old program that claims to read AWS tapes but doesn't read `simh` AWS tape images.  But let's not go there yet!*

**Comment by markpizz** [comment]
Well, when I implemented this support, I worked from what I could find which was more or less informally described.  I don't specifically recall.  I then tested the functionality that was implemented with any tape images I could find and didn't have any problems with those files.

What is the purpose of this "old program that claims to read AWS tapes but doesn't read simh AWS tape images"?

**Comment by gavineadie** [comment]
I’m glad to say this turns out to be the “simple” variation. For context, I and a few friends have been archiving the time sharing system (MTS) we provided at the university of michigan till about 1996. Part of that project was resurrecting the OS on Hercules, and we provided distribution tapes etc to bitsavers as AWS tapes. Those were IBM labeled “tapes” created by a program we wrote a long time ago. That program had not been much exposed to ANSI labeled tapes images and the images I was generating with simh were ANSI labeled. It turns out there was a bug in our processing of ANSI tapes, now fixed.

In passing, I’ll note that the past few weeks have seen another association between our project and simh via a request for us to explore bringing up a ~50 year old version of MTS on a simh-based IBM 360/67 emulator. I expect AWS tapes and CKD disks will feature in this work.
